stubs/tensorflow/tensorflow/__init__.pyi

```diff
@@ -0,0 +1,195 @@
+from _typeshed import Incomplete, Self, Unused
+from abc import ABCMeta
+from builtins import bool as _bool
+from collections.abc import Callable, Iterable, Iterator, Sequence
+from contextlib import contextmanager
+from enum import Enum
+from typing import Any, NoReturn, overload
+from typing_extensions import TypeAlias
+
+import numpy as np
+from tensorflow.dtypes import *
+
+# Most tf.math functions are exported as tf, but sadly not all are.
+from tensorflow.math import abs as abs
+from tensorflow.sparse import SparseTensor
+
+# Tensors ideally should be a generic type, but properly typing data type/shape
+# will be a lot of work. Until we have good non-generic tensorflow stubs,
+# we will skip making Tensor generic. Also good type hints for shapes will
+# run quickly into many places where type system is not strong enough today.
+# So shape typing is probably not worth doing anytime soon.
+_Slice: TypeAlias = int | slice | None
+
+_FloatDataSequence: TypeAlias = Sequence[float] | Sequence[_FloatDataSequence]
+_StrDataSequence: TypeAlias = Sequence[str] | Sequence[_StrDataSequence]
+_ScalarTensorCompatible: TypeAlias = Tensor | str | float | np.ndarray[Any, Any] | np.number[Any]
+_TensorCompatible: TypeAlias = _ScalarTensorCompatible | Sequence[_TensorCompatible]
+_ShapeLike: TypeAlias = TensorShape | Iterable[_ScalarTensorCompatible | None] | int | Tensor
+_DTypeLike: TypeAlias = DType | str | np.dtype[Any]
+
+class Tensor:
+    def __init__(self, op: Operation, value_index: int, dtype: DType) -> None: ...
```

Comment:
I'm not sure why I need return type for this `__init__` (and a couple others). Without it pyright in CI shows an error message of unknown return type. Locally I can't reproduce that error.

Reply:
We have strict pyright settings in CI that disallow leaving any parameters or returns without type annotations. You can switch off the strict settings by adding `stubs/tensorflow` to the excludelist here:

https://github.com/python/typeshed/blob/5eb6a692f5e52ad374b8ab5434a351a149cf02ce/pyrightconfig.stricter.json#L8

But we prefer it if as many stubs as possible are tested with the stricter settings!

Reply:
Keeping it to pyright strict sounds good to me.